dandi/validate.py

```diff
@@ -1,11 +1,42 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from enum import Enum
 from pathlib import Path
 from typing import Iterator, List, Optional, Tuple, Union
 
 import appdirs
 
 from .files import find_dandi_files
 
-# TODO: provide our own "errors" records, which would also include warnings etc
+
+@dataclass
+class ValidationResult:
+    origin: ValidationOrigin
+    severity: Severity
+    id: str
+    scope: Scope
+    path: Path
+    message: str
+    dataset_path: Optional[Path]
+    asset_paths: Optional[list[str]] = None
+
+
+@dataclass
+class ValidationOrigin:
+    name: str
+    version: str
+
+
+class Severity(Enum):
+    ERROR = "ERROR"
+    WARNING = "WARNING"
+    HINT = "HINT"
```

Comment:
Interesting, what are some examples you can think of for `HINT`s? Are these asset-level, like minor style suggestions for NWBFile metadata?

Reply:
For nwb within BIDS dataset might potentially be something like "place different modalities into separate .nwb files". I could come up with others later ;)

Reply:
as I have provided an example, I will consider this one resolved ;)